master/internal/stream/trials.go

```diff
@@ -216,3 +237,21 @@ func (ts *TrialFilterMaker) MakeFilter() func(*TrialMsg) bool {
 		return false
 	}
 }
+
+// TrialMakePermissionFilter returns a function that checks if a TrialMsg
+// is in scope of the user permissions.
+func TrialMakePermissionFilter(ctx context.Context, user model.User) (func(*TrialMsg) bool, error) {
+	accessScopeSet, err := AuthZProvider.Get().GetTrialStreamableScopes(ctx, user)
+	if err != nil {
+		return nil, err
+	}
+
+	return func(msg *TrialMsg) bool {
+		if _, ok := accessScopeSet[model.GlobalAccessScopeID]; ok {
+			return true
```

Comment:
Can we detect if this is the case, and return a `func(msg *TrialMsg){ return true }` closure instead?

No point in doing thousands of dictionary lookups when one will suffice.

Reply:
Good call!